gather_istio.sh

```diff
@@ -0,0 +1,224 @@
+      oc exec -n "${podNamespace}" "${podName}" -c istio-proxy -- /usr/local/bin/pilot-agent request GET config_dump > "${logPath}/config_dump_proxy.json" 2>&1
+      oc exec -n "${podNamespace}" "${podName}" -c istio-proxy -- /usr/local/bin/pilot-agent request GET stats > "${logPath}/proxy_stats" 2>&1
+    fi
+  done
+}
+
+function version() {
+  if [[ -n $OSSM_MUST_GATHER_VERSION ]] ; then
+    echo "${OSSM_MUST_GATHER_VERSION}"
+  else
+    echo "0.0.0-unknown"
+  fi
+}
+
+function inspect() {
+  local resource ns
+  resource=$1
+  ns=$2
+
+  echo
+  if [ -n "$ns" ]; then
+    echo "Inspecting resource ${resource} in namespace ${ns}"
+    oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+  else
+    echo "Inspecting resource ${resource}"
+    oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+  fi
+}
+
+function inspectNamespace() {
+  local ns
+  ns=$1
+
+  inspect "ns/$ns"
+  for crd in $crds; do
+    inspect "$crd" "$ns"
+  done
+  inspect net-attach-def,roles,rolebindings "$ns"
+}
+
+function main() {
+  #TODO: respect --since and --since-time: https://github.com/openshift/enhancements/blob/master/enhancements/oc/must-gather.md#must-gather-images
+  local crds controlPlanes
+  echo
+  echo "Executing Istio gather script"
+  echo
+
+  versionFile="${BASE_COLLECTION_PATH}/version"
+  echo "openshift-service-mesh/must-gather"> "$versionFile"
+  version >> "$versionFile"
+
+  # TODO: add new name label to servicemesh-operator3 pod and use that instead of the control-plane label
+  operatorNamespace=$(oc get pods --all-namespaces -l control-plane=servicemesh-operator3 -o jsonpath="{.items[0].metadata.namespace}")
+
+  inspect nodes
+
+  # TODO: does this match everything we need?
+  for r in $(oc get clusterroles,clusterrolebindings -l install.operator.istio.io/owning-resource -oname); do
+    inspect "$r"
+  done
+  for r in $(oc get clusterroles,clusterrolebindings -l 'app in (istiod,istio-reader)' -oname); do
+    inspect "$r"
+  done
+
+
+  crds="$(getCRDs)"
+  for crd in ${crds}; do
+    inspect "${crd}"
+  done
+
+  controlPlanes="$*"
+  if [ -z "${controlPlanes}" ]; then
+    controlPlanes="$(getControlPlanes)"
+  fi
+
+  inspect "ns/$operatorNamespace"
+  inspect clusterserviceversion "${operatorNamespace}"
+
+  # inspect all controlled mutatingwebhookconfiguration
+  for mwc in $(oc get mutatingwebhookconfiguration -l app=sidecar-injector);do
+    inspect ${mwc}
+  done
+  # inspect all controlled validatingwebhookconfiguration
+  for vwc in $(oc get validatingwebhookconfiguration -l app=istiod);do
+    inspect ${vwc}
+  done
+
+  for cp in ${controlPlanes}; do
+      echo
+      echo "Processing control plane namespace: ${cp}"
+
+      crds="$crds" inspectNamespace "$cp"
+      #getEnvoyConfigForPodsInNamespace "${cp}" "${cp}"
+      getSynchronization "${cp}"
+
+      #TODO: how to get proxy <> control plane mapping? Using istioctl proxy-status? Using just namespaces labels
```

Comment:
Namespace labels and pod labels (labels `istio-injection: true` and `istio.io/rev: <rev-name>`.

Perhaps we can also use istioctl, yes.